### PR TITLE
fix : azure-messaging-eventhubs test cases

### PR DESCRIPTION
**Pull request by gkanani**
fix : #6479

### CODE REVIEW

File: sdk/eventhubs/azure-messaging-eventhubs/pom.xml [hunk 1]
```diff
@@ -104,6 +104,16 @@
...
+
+      <plugin>
+        <groupId>org.revapi</groupId>
+        <artifactId>revapi-maven-plugin</artifactId>
+        <version>0.11.2</version> <!-- {x-version-update;org.revapi:revapi-maven-plugin;external_dependency} -->
+        <configuration>
+          <skip>true</skip>
+        </configuration>
+      </plugin>
+
```
Comment (srnagar) [inline sdk/eventhubs/azure-messaging-eventhubs/pom.xml:116]: Revapi checks should not be skipped. This plugin can be removed.

File: sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/implementation/EventHubReactorConnectionTest.java [hunk 2]
```diff
@@ -158,6 +160,13 @@ public void getsManagementChannel() {
         StepVerifier.create(connection.getManagementNode())
             .assertNext(node -> Assertions.assertTrue(node instanceof ManagementChannel))
             .verifyComplete();
+
+        StepVerifier.create(connection.createSendLink("send-name", "test-entity-path", new AmqpRetryOptions()))
+            .verifyError();
+
+        StepVerifier.create(connection.createReceiveLink("receiver-name", "test-entity-path", EventPosition.latest(), new ReceiveOptions()))
+            .verifyError();
```
Comment (srnagar) [inline sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/implementation/EventHubReactorConnectionTest.java:168]: What's the error we are verifying here? Can we also assert on the expected error?

File: sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/implementation/EventHubConnectionProcessorTest.java [hunk 3]
```diff
@@ -286,6 +286,24 @@ void requiresNonNull() {
...
+    @Test
+    public void getEventHubNameTest() {
+        String testEventHubName = eventHubConnectionProcessor.getEventHubName();
+        Assertions.assertEquals("test-event-hub-name", testEventHubName);
+    }
+
+    @Test
+    public void getRetryOptionsTest() {
+        AmqpRetryOptions testRetryOption = eventHubConnectionProcessor.getRetryOptions();
+        Assertions.assertNotNull(testRetryOption);
```
Comment (srnagar) [inline sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/implementation/EventHubConnectionProcessorTest.java:304]: Assert the values of retry mode, maxDelay, maxRetries etc as well.

**Review by srnagar — COMMENTED** [review]
@gkanani Thank you for your contribution. We want to keep the revapi checks for this package. So, the plugin configuration to skip this check should be removed from pom.xml.

**Comment by gkanani** [comment]
@srnagar  PTAL

File: sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/implementation/EventHubReactorConnectionTest.java [hunk 4]
```diff
@@ -158,6 +160,15 @@ public void getsManagementChannel() {
         StepVerifier.create(connection.getManagementNode())
             .assertNext(node -> Assertions.assertTrue(node instanceof ManagementChannel))
             .verifyComplete();
+
+        System.out.println("-----------------" + connection.createSendLink("send-name", "test-entity-path", new AmqpRetryOptions()));
```
Comment (srnagar) [inline sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/implementation/EventHubReactorConnectionTest.java:164]: Remove print statement here.

File: sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/implementation/EventHubReactorConnectionTest.java [hunk 5]
```diff
@@ -158,6 +160,15 @@ public void getsManagementChannel() {
...
+
+        StepVerifier.create(connection.createSendLink("send-name", "test-entity-path", new AmqpRetryOptions()))
+            .verifyError(NullPointerException.class);
+
+        StepVerifier.create(connection.createReceiveLink("receiver-name", "test-entity-path", EventPosition.latest(), new ReceiveOptions()))
+            .verifyError(NullPointerException.class);
```
Comment (srnagar) [inline sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/implementation/EventHubReactorConnectionTest.java:170]: I don't think the mocks are set up correctly for testing `createSendLink` and `createReceiveLink` - `NullPointerException` is not the expected behavior here. Also, these have to be in separate test methods as this method is for testing management channel. Send and receive should be in its own test methods.

File: sdk/eventhubs/azure-messaging-eventhubs/pom.xml [hunk 6]
```diff
@@ -104,6 +104,8 @@
           </systemPropertyVariables>
         </configuration>
       </plugin>
+
+
```
Comment (srnagar) [inline sdk/eventhubs/azure-messaging-eventhubs/pom.xml:108]: nit: remove extra blank lines.

**Comment by gkanani** [comment]
@srnagar PTAL

File: sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/implementation/EventHubReactorConnectionTest.java [hunk 7]
```diff
@@ -158,6 +154,24 @@ public void getsManagementChannel() {
...
             .verifyComplete();
+
+    }
+
+    @Test
+    public void createSendLinkTest() {
+        final EventHubReactorAmqpConnection connection = new EventHubReactorAmqpConnection(CONNECTION_ID,
+            connectionOptions, "event-hub-name", reactorProvider, handlerProvider, tokenManagerProvider,
+            messageSerializer, product, clientVersion);
+        Assertions.assertNotNull(connection.createSendLink("send-name", "test-entity-path", new AmqpRetryOptions()));
```
Comment (srnagar) [inline sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/implementation/EventHubReactorConnectionTest.java:165]: `createSendLink` is an async operation and will always return a non-null `Mono` object which should be subscribed to verify that the link was created. See the unit test for `getsManagementChannel`. Something similar to this has to be done:
```java
 StepVerifier.create(connection.createSendLink())
            .assertNext(link -> Assertions.assertTrue(link instanceof ReactorSender))
            .verifyComplete();
```

File: sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/implementation/EventHubReactorConnectionTest.java [hunk 8]
```diff
@@ -158,6 +154,24 @@ public void getsManagementChannel() {
...
+    }
+
+    @Test
+    public void createReceiveLinkTest() {
+        final EventHubReactorAmqpConnection connection = new EventHubReactorAmqpConnection(CONNECTION_ID,
+            connectionOptions, "event-hub-name",
+            reactorProvider, handlerProvider, tokenManagerProvider,
+            messageSerializer, product, clientVersion);
+        Assertions.assertNotNull(connection.createReceiveLink("receiver-name", "test-entity-path", EventPosition.latest(), new ReceiveOptions()));
```
Comment (srnagar) [inline sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/implementation/EventHubReactorConnectionTest.java:174]: Same as above.

**Review by srnagar — CHANGES_REQUESTED** [review]

**Pull request closed without merging**